app/models/concerns/foreman_kernel_care/host_managed_extensions.rb

```diff
@@ -29,26 +28,56 @@ def import_tracer_profile(tracer_profile)
       end
     end
 
-    def update_kernel_version(version, release)
-      packages = ::Katello::InstalledPackage.where('name LIKE ?', '%kernel%').where.not(name: 'kernelcare').to_a
-      delete_ids = []
-      simple_packages = packages.map do |p|
-        delete_ids << p.id
-        ::Katello::Pulp::SimplePackage.new({
-                                             arch: p.arch,
-                                             epoch: p.epoch,
-                                             version: version,
-                                             release: release,
-                                             name: p.name
-                                           })
+    def update_kernel_version(version, release, old_kernel_version)
+      old_version, old_release = old_kernel_version.split('-')
+
+      case operatingsystem.family
+      when 'Redhat'
+        packages = find_yum_kernel_packages(old_version, old_release)
+        return if packages.empty?
+
+        create_new_yum_kernel_packages(packages, version, release)
```

Comment:
It would be more intuitive, if the create_new_yum_kernel_packages (and the create_new_deb_kernel_packages) only create the new package and association. 
To delete the old assoc it should be done before so that it looks like:
```
pkg = find_packages()
delete_pkgs(pkgs)
create_new_pkgs()
```

Reply:
kernelcare "kernels" cleaning up only if kernelcare tool was deleted, and the user install something via yum/apt
You are right, we should think about it in the future

> a brand new kernel is installed and the kernel-care kernel is no longer necessary

We decided to leave kernelcare "kernels" in that case because it's really rare

Reply:
I mean, the kernelcare kernels in katello db for the host are not cleaned up.